### ISSUE TITLE
Lot of mutateAsync functions not being awaited

### ISSUE THREAD

**Issue by harshsbhat** [unkeyed/unkey#2205]
### Preliminary Checks

- [X] I have reviewed the documentation: https://unkey.com/docs

- [X] I have searched for existing issues: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Unkey. Please ask questions in our Discord community: https://unkey.com/discord.

### Issue Summary

It is usually a good practice to await the mutation async functions. We are missing that in a lot of instances.  I have just listed one of them. There are at least a few more of these being used without awaiting 

```
  async function onSubmit(values: z.infer<typeof formSchema>) {
    updateRatelimit.mutateAsync(values);
  }
```

### Steps to Reproduce

NA

### Expected behavior

We should await these functions. 

### Version info

```markdown
- OS:
- Node:
- npm:
```

**Comment by NeerajReddyK** [comment]
Can I get that assigned? (if it is up for grabs)

**Comment by harshsbhat** [comment]
> /assign

I was testing it and was done with it. If you have started working I can discard the changes. But if you `haven't`  can you stop?

**Comment by DeepaPrasanna** [comment]
> > /assign
> 
> I was testing it and was done with it. If you have started working I can discard the changes. But if you `haven't` can you stop?

Hey, @chronark will I be wrong in saying that one should start working on an issue only after being assigned, so that others wont have any misunderstanding. I have seen in other oss.gg repos that you cant work on more than 1 issue in the same repo at the same time. You can move to another issue only after your previous issue has been merged/closed so that this gives everyone a fair chance. Apologies if I misunderstood/or spoke wrong 🙏

**Comment by harshsbhat** [comment]
> > > /assign
> > 
> > 
> > I was testing it and was done with it. If you have started working I can discard the changes. But if you `haven't` can you stop?
> 
> Hey, @chronark will I be wrong in saying that one should start working on an issue only after being assigned, so that others wont have any misunderstanding. I have seen in other oss.gg repos that you cant work on more than 1 issue in the same repo at the same time. You can move to another issue only after your previous issue has been merged/closed so that this gives everyone a fair chance. Apologies if I misunderstood/or spoke wrong 🙏

The thing is my other PR ( which I was assigned is already approved and tests are running ) that's why I asked him. Hope that was fair. I apologies it was fair.

**Comment by DeepaPrasanna** [comment]
I think you should wait for your PR to be closed/merged. You can try /assign here. I believe it wont work when you have open PRs

**Comment by chronark** [comment]
Hmm, yeah we should try to distribute issues a bit better, so newcomers have a chance to grab some too.
The "only one active PR" rule is a bit too strict sometimes, cause that means you're potentially blocked on my review.

**Comment by DeepaPrasanna** [comment]
Yes, I do agree. Yesterday, I had to wait until my PR got merged to assign myself another(I understand It is not easy you guys do reviews and work on new features at the same time. I am grateful for all the reviews that I recieve.) but I believe we can use that time to explore other repos or try non code contributions.

**Comment by harshsbhat** [comment]
should I open the PR for this one or not then? 

@chronark @DeepaPrasanna I think having a tag like `first-timers only` like  freecodecamp can help us give new comers a good and fair chance on the simpler issues wdyt?

**Comment by chronark** [comment]
let's leave it open for someone else to come in
if nobody takes it within the next 24h, go ahead

**Comment by harshsbhat** [comment]
@geraldmaboshe  feel free to refer my closed PR

**Comment by geraldmaboshe** [comment]
cool, thanks for the cheatsheet @harshsbhat 😃

**Comment by harshsbhat** [comment]
@anand1564 there is already a PR open for this issue
